src/frontend/planner_test/tests/testdata/output/sink.yaml

```diff
@@ -172,3 +172,45 @@
     StreamSink { type: upsert, columns: [a, b, t2._row_id(hidden)], pk: [t2._row_id] }
     └─StreamEowcSort { sort_column: t2.b }
       └─StreamTableScan { table: t2, columns: [a, b, _row_id] }
+- id: create_mock_iceberg_sink_append_only_with_partition
```

Comment:
This test cases seem not enough, in the rfc there are 4 cases, please add test for each case.